applications/credhub-api/src/main/resources/db/migration/mysql/V57_2__delete_related_encrypted_value_when_credential_is_deleted.sql

```diff
@@ -0,0 +1,50 @@
+-- Fix for https://github.com/cloudfoundry/credhub/issues/231
+-- Create delete trigger on credential table instead of the tables directly
+-- reference encrypted_value table because the trigger does not work for
+-- cascade-deleted records in MySQL.
+DROP TRIGGER IF EXISTS tr_credential_deleted;
+DELIMITER |
+CREATE TRIGGER tr_credential_deleted BEFORE DELETE ON credential
+FOR EACH ROW
+BEGIN
+  DROP TEMPORARY TABLE IF EXISTS
+    tmp_credential_version_uuids, tmp_encrypted_value_uuids;
+
+  -- Remember the uuids of the credential_version records being deleted
+  CREATE TEMPORARY TABLE tmp_credential_version_uuids
+    SELECT uuid FROM credential_version
+      WHERE credential_uuid = OLD.uuid;
```

Comment:
I was hoping this could be done without creating temporary tables, this looks pretty heavy weight for what we are trying to do.

Is it really our only option to create 2 temporary tables to store one ID in each?

Reply:
Temporary tables are in-memory tables and not too heavy weight. That is only proper way to that can be done in MySQL that I know of.

Reply:
Temp table use could be avoided if we delete encrypted_value records that are not referenced by those 3 tables anymore after the those 3 table records were deleted. However, the performance will be worse that way because you now have to do full scan of the 3 tables every time.

Reply:
I am going to rewrite it not to use temp tables. There are potential issues with using temp tables inside trigger as we do not have complete control of how/when the trigger is executed.